### PR TITLE
panic instead of error when decrease counter

### PR DESCRIPTION
**Pull request by breeswish**
Fix https://github.com/pingcap/rust-prometheus/issues/151

this simplifies and unifies the counter interface.

because it is a breaking change, a changelog file is introduced.

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

**Comment by breeswish** [comment]
@overvenus PTAL

File: src/counter.rs [hunk 1]
```diff
@@ -138,15 +137,14 @@ impl LocalCounter {
...
-    /// `inc_by` increments the given value to the local counter. Error if the value is <
-    /// 0.
+    /// `inc_by` increments the given value to the local counter. Panics if
+    /// the value is < 0.
     #[inline]
-    pub fn inc_by(&mut self, v: f64) -> Result<()> {
+    pub fn inc_by(&mut self, v: f64) {
         if v < 0.0 {
-            return Err(Error::DecreaseCounter(v));
+            panic!("counter cannot inc negative values")
```
Comment (overvenus) [inline src/counter.rs]: Could you test panic?

File: src/counter.rs [hunk 2]
```diff
@@ -138,15 +137,14 @@ impl LocalCounter {
         }
     }
 
-    /// `inc_by` increments the given value to the local counter. Error if the value is <
-    /// 0.
+    /// `inc_by` increments the given value to the local counter. Panics if
+    /// the value is < 0.
```
Comment (overvenus) [inline src/counter.rs]: Could you document it as a panic section? 
https://doc.rust-lang.org/book/first-edition/documentation.html#special-sections

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Changelog
+
+### 0.4.0
```
Comment (overvenus) [inline CHANGELOG.md]: `## 0.4.0`

**Comment by breeswish** [comment]
@overvenus Done, PTAL

**Comment by siddontang** [comment]
CI failed.

**Comment by breeswish** [comment]
@siddontang It's a Travis CI service failure. Triggered rebuild to see if it is ok now.

**Comment by breeswish** [comment]
Build passed. Merged.

**Comment by breeswish** [comment]
Oh. I should use squash merge : (

**Pull request merged**